File: examples/dr_wav_simple.c
#define DR_WAV_IMPLEMENTATION
#include <stdio.h>
#include "dr_wav.h"


int main(int argc, char* argv[]) {
    drwav wav;
    drwav_fmt fmt;

    if(argc < 2){
        printf("usage: dr_wav_simple <WAV file>\n");
        return 1;
    }

    if (!drwav_init_file(&wav, argv[1], NULL)) {
        return -1;
    }

    printf("'%s' has %d sample frames.\n", argv[1], wav.totalPCMFrameCount);
    printf("Each using %d channel(s).\n", wav.channels);
    printf("The sample rate is %d Hz.\n", wav.sampleRate);

    int32_t* pSampleData = (int32_t*)malloc((size_t)wav.totalPCMFrameCount * wav.channels * sizeof(int32_t));
    size_t num_samples = drwav_read_pcm_frames_s32(&wav, wav.totalPCMFrameCount, pSampleData);

    // At this point pSampleData contains every decoded sample as signed 32-bit PCM.
    // Channels are interleaved.

    drwav_uninit(&wav);
    return 0;
}

File: examples/dr_wav_simple.py
#!/usr/bin/env python

import sys

from dr_libs import dr_wav


def main():
    if len(sys.argv) < 2:
        return "usage: dr_wav_simple.py <WAV file>"

    with dr_wav.DrWav(sys.argv[1]) as wav:
        print("'%s' has %d sample frames." % (sys.argv[1], wav.nframes))
        print("Each using %d channel(s)." % wav.channels)
        print("The sample rate is %d Hz." % wav.sample_rate)

        samples = wav.read()  # noqa:F841

        # At this point samples is an array.array('i') instance,
        # which contains wav.channels * wav.nframes samples as signed 32-bit PCM.
        # Channels are interleaved.

    # Exiting the context calls wav.close() implicitly.
    # It is also automatically called when 'wav' is garbage-collected


if __name__ == "__main__":
    sys.exit(main() or 0)
